internal/servers/job/rotate_roots_job_test.go

```diff
@@ -0,0 +1,106 @@
+package servers
+
+import (
+	"context"
+	"crypto/rand"
+	"testing"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/servers"
+	"github.com/hashicorp/boundary/internal/types/scope"
+	"github.com/hashicorp/nodeenrollment/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRotateRootsJob(t *testing.T) {
+	require, assert := require.New(t), assert.New(t)
+	ctx := context.Background()
+	wrapper := db.TestWrapper(t)
+	conn, _ := db.TestSetup(t, "postgres")
+	rw := db.New(conn)
+	kmsCache := kms.TestKms(t, conn, wrapper)
+	err := kmsCache.CreateKeys(context.Background(), scope.Global.String(), kms.WithRandomReader(rand.Reader))
+	require.NoError(err)
+	wrapper, err = kmsCache.GetWrapper(context.Background(), scope.Global.String(), kms.KeyPurposeDatabase)
+	require.NoError(err)
+	require.NotNil(wrapper)
+
+	workerAuthRepo, err := servers.NewRepositoryStorage(ctx, rw, rw, kmsCache)
+	require.NoError(err)
+
+	type args struct {
+		w   db.Writer
+		r   db.Reader
+		kms *kms.Kms
+	}
+	tests := []struct {
+		name        string
+		args        args
+		options     []servers.Option
+		wantLimit   int
+		wantErr     bool
+		wantErrCode errors.Code
+	}{
+		{
+			name: "nil writer",
+			args: args{
+				r:   rw,
+				kms: kmsCache,
+			},
+			wantErr:     true,
+			wantErrCode: errors.InvalidParameter,
+		},
+		{
+			name: "nil reader",
+			args: args{
+				w:   rw,
+				kms: kmsCache,
+			},
+			wantErr:     true,
+			wantErrCode: errors.InvalidParameter,
+		},
+		{
+			name: "nil kms",
+			args: args{
+				w: rw,
+				r: rw,
+			},
+			wantErr:     true,
+			wantErrCode: errors.InvalidParameter,
+		},
+		{
+			name: "valid",
+			args: args{
+				w:   rw,
+				r:   rw,
+				kms: kmsCache,
+			},
+			wantLimit: db.DefaultLimit,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := newRotateRootsJob(ctx, tt.args.r, tt.args.w, tt.args.kms)
+			if tt.wantErr {
+				require.Error(err)
+				assert.Nil(got)
+				assert.Truef(errors.Match(errors.T(tt.wantErrCode), err), "Unexpected error %s", err)
+				return
+			}
+			require.NoError(err)
+			require.NotNil(got)
+			require.Equal(0, got.totalRotates)
+
+			// Run job and ensure rotation was performed
```

Comment:
minor nit: As I am not sure exactly what validation for rotate means. It might be nice to see a `workerAuthRepo.List(` before `got.Run` to see to that the List moved from x to `2`